File: go-client-sdk/pkg/models/operations/getcompaniescompanyiddatataxrates.go
package operations

import (
	"github.com/speakeasy-sdks/codat-sdks/go-client-sdk/v2/pkg/models/shared"
)

type GetCompaniesCompanyIDDataTaxRatesPathParams struct {
	CompanyID string `pathParam:"style=simple,explode=false,name=companyId"`
}

type GetCompaniesCompanyIDDataTaxRatesQueryParams struct {
	OrderBy  *string `queryParam:"style=form,explode=true,name=orderBy"`
	Page     int     `queryParam:"style=form,explode=true,name=page"`
	PageSize *int    `queryParam:"style=form,explode=true,name=pageSize"`
	Query    *string `queryParam:"style=form,explode=true,name=query"`
}

type GetCompaniesCompanyIDDataTaxRatesSecurity struct {
	APIKeyAuth *shared.SchemeAPIKeyAuth `security:"scheme,type=apiKey,subtype=header"`
	CodatLogin *shared.SchemeCodatLogin `security:"scheme,type=oauth2"`
}

type GetCompaniesCompanyIDDataTaxRatesRequest struct {
	PathParams  GetCompaniesCompanyIDDataTaxRatesPathParams
	QueryParams GetCompaniesCompanyIDDataTaxRatesQueryParams
	Security    GetCompaniesCompanyIDDataTaxRatesSecurity
}

type GetCompaniesCompanyIDDataTaxRatesResponse struct {
	CodatDataContractsDatasetsTaxRatePagedResponseModel *shared.CodatDataContractsDatasetsTaxRatePagedResponseModel
	ContentType                                         string
	StatusCode                                          int
}

File: typescript-client-sdk/src/sdk/models/operations/getcompaniescompanyiddatataxrates.ts
import { SpeakeasyBase, SpeakeasyMetadata } from "../../../internal/utils";
import * as shared from "../shared";
import { Type } from "class-transformer";


export class GetCompaniesCompanyIdDataTaxRatesPathParams extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "pathParam, style=simple;explode=false;name=companyId" })
  companyId: string;
}

export class GetCompaniesCompanyIdDataTaxRatesQueryParams extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "queryParam, style=form;explode=true;name=orderBy" })
  orderBy?: string;

  @SpeakeasyMetadata({ data: "queryParam, style=form;explode=true;name=page" })
  page: number;

  @SpeakeasyMetadata({ data: "queryParam, style=form;explode=true;name=pageSize" })
  pageSize?: number;

  @SpeakeasyMetadata({ data: "queryParam, style=form;explode=true;name=query" })
  query?: string;
}

export class GetCompaniesCompanyIdDataTaxRatesSecurity extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "security, scheme=true;type=apiKey;subtype=header" })
  apiKeyAuth?: shared.SchemeAPIKeyAuth;

  @SpeakeasyMetadata({ data: "security, scheme=true;type=oauth2" })
  codatLogin?: shared.SchemeCodatLogin;
}

export class GetCompaniesCompanyIdDataTaxRatesRequest extends SpeakeasyBase {
  @SpeakeasyMetadata()
  pathParams: GetCompaniesCompanyIdDataTaxRatesPathParams;

  @SpeakeasyMetadata()
  queryParams: GetCompaniesCompanyIdDataTaxRatesQueryParams;

  @SpeakeasyMetadata()
  security: GetCompaniesCompanyIdDataTaxRatesSecurity;
}

export class GetCompaniesCompanyIdDataTaxRatesResponse extends SpeakeasyBase {
  @SpeakeasyMetadata()
  codatDataContractsDatasetsTaxRatePagedResponseModel?: shared.CodatDataContractsDatasetsTaxRatePagedResponseModel;

  @SpeakeasyMetadata()
  contentType: string;

  @SpeakeasyMetadata()
  statusCode: number;
}